Sources/Quick/StopTest.swift

```diff
@@ -0,0 +1,69 @@
+/// Stops execution of test when thrown inside an `it` block, emitting a failure message.
+///
+/// Unlike other errors that can be thrown during a test, `StopTest` represents an expected
+/// failure, with the failure description tied to the file and line it is thrown on.
+///
+/// Tests can also be stopped silently by throwing `StopTest.silently`.
+///
+/// The primary use case of `StopTest` as opposed to normal error logging is when a condition
+/// is critical for the remainder of the test.  It serves as an alternative to force unwrapping
+/// or out-of-range subscripts that could be cause the test to crash.
+///
+/// For example,
+///
+/// ```
+/// guard let value = getValue() else {
+///     throw StopTest("Got a null value from `getValue()`)
+/// }
+/// ```
+///
+/// When used with Nimble, any expectation can stop a test by adding
+/// `.onFailure(throw: StopTest.silently)`.
+///
+/// For example,
+///
+/// ```
+/// try expect(array).toEventually(haveCount(10)).onFailure(throw: StopTest.silently)
+/// ```
+public struct StopTest: Error {
+    public let failureDescription: String
+    public let reportError: Bool
+    public let callsite: Callsite
+
+    private init(_ failureDescription: String, reportError: Bool, file: FileString, line: UInt) {
```

Comment:
This needs to be public to allow it to actually build and be used.

```suggestion
    public init(_ failureDescription: String, reportError: Bool, file: FileString, line: UInt) {
```

Reply:
This initializer is supposed to be private, with the public initializer right below it.  The reason I added it was so there's exactly one place to create a non-reported error, `StopTest.silently`.  I added a comment to describe its purpose.

The `rake` failures were because I didn't add StopTest.swift to the xcproject.

Reply:
> The rake failures were because I didn't add StopTest.swift to the xcproject.

Whoops. Thanks for noticing that oversight.